### ISSUE TITLE
Add directions for Vue SFC template and script linting

### ISSUE THREAD

**Issue by IlyaSemenov** [fimbullinter/wotan#429]
There is a whole subproject Vé dedicated to first class Wotan Vue SFC linting, however there isn't a word about linting SFC template and script parts. I understand that it's not this project's direct goal, however I believe giving recommended directions and/or explaining best practices would be useful to most Vue users coming for Vé.

For example, right now I am using Wotan and Valtyr for scripts and eslint with typescript-eslint-parser and eslint-plugin-vue for Vue templates (and Vue specific script checks). Is that the best possible approach at the moment, or is there a simpler/better method (ideally calling everything from a single `wotan` run)?

**Comment by ajafff** [comment]
It wasn't immediately apparent to me that "template linting" means you want to analyse the content of `<template></template>`.
This is currently not possible as the processor only extracts the `<script>` part of the SFC and passes that to the linter. After all this linter is intended for JavaScript and TypeScript.

---

Nevertheless, it could be possible to implement.
Some research revealed that SFC files can roughly be converted to pure JS as follows (please correct me if I'm wrong):

```vue
<template>
  <p>{{ greeting }} World!</p>
</template>
<script>
export default {
  data() { return { greeting: 'Hello' }; },
};
</script>
```

```js
export default {
  data() { return { greeting: 'Hello' }; },
  template: '<p>{{ greeting }} World!</p>',
};
```

That means it could be possible to pass the template content as part of the script to the linter. There only needs to be a rule to analyse that specific part. The processor would then map the error locations back to the original SFC format (kind of how source maps work).

**Comment by IlyaSemenov** [comment]
Yes, that's what I mean, lint the contents of the templates. I'm currently using `eslint-plugin-vue` for that (example of the linting rule: https://github.com/vuejs/eslint-plugin-vue/blob/master/docs/rules/html-indent.md). I am not linting SFC style blocks at all, even though it's also desired.

As I mentioned above, I understand that this is a JS/TS linter and linting Vue HTML not its direct business. That's why it's not a request for implementation but rather just for documenting suggested solutions and generic dos/donts.

---

By the way, Vue templates *may* include certain JS/TS code. For example, in a Vue template like `<div>{{ obj.key }}</div>` ideally `obj` should be validated to: 1) exist in the Vue component `props`/`data`/`computed` 2) have `key` property.
